domain/sysvar_cache.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package domain
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/util/logutil"
+	"github.com/pingcap/tidb/util/sqlexec"
+	"github.com/pingcap/tidb/util/stmtsummary"
+	"go.uber.org/zap"
+)
+
+// The sysvar cache replaces the GlobalVariableCache.
+// It is an improvement because it operates similar to privilege cache,
+// where it caches for 5 minutes instead of 2 seconds, plus it listens on etcd
+// for updates from other servers.
+
+// SysVarCache represents the cache of system variables broken up into session and global scope.
+type SysVarCache struct {
+	sync.RWMutex
+	global  map[string]string
+	session map[string]string
+}
+
+// GetSysVarCache gets the global variable cache.
+func (do *Domain) GetSysVarCache() *SysVarCache {
+	return &do.sysVarCache
+}
+
+func (svc *SysVarCache) rebuildCacheIfNeeded(ctx sessionctx.Context) {
```

Comment:
No return result to indicate if it success?

Reply:
This is addressed in https://github.com/pingcap/tidb/pull/24359/commits/3be42568ebeb33bd0c993ba486ad0a4fa1642545